### PR TITLE
Added Annual report tests

### PR DESCRIPTION
**Pull request by vovikboyko**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 1]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(actual, expected);
+            Assert.IsInstanceOf<ObjectResult>(result);
+
+            var statuses = new List<string>();
+            foreach (var enumValue in Enum.GetValues(typeof(AnnualReportStatusDTO)).Cast<AnnualReportStatusDTO>())
+            {
+                statuses.Add(enumValue.GetDescription());
+            }
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:899]: I do not understand this?

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 2]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+
+            //TO DO: ensure, that result is AnnualReportStatusDTO type
+            //var w = (result as ObjectResult).Value;
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:902]: ?

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 3]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+        }
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:905]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 4]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var expected = StatusCodes.Status200OK;
+            var actual = (result as ObjectResult).StatusCode;
+
+            //Assert
+            Assert.AreEqual(expected, actual);
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<ObjectResult>(result);
+        }
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:926]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 5]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(expected, actual);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+
+        }
+
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:989]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 6]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var actual = (result as StatusCodeResult).StatusCode;
+
+
+            // Assert
+            Assert.AreEqual(expected, actual);
+            Assert.NotNull(result);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+        }
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1071]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 7]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var result = await annualReportController.CancelClubAnnualReport(5);
+            var actual = (result as StatusCodeResult).StatusCode;
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(expected, actual);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+        }
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1196]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 8]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1135]: unnecessary things

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs (hunk 2 above)
Comment (LanchevychMaxym) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:902]: Coments

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 9]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var result = await annualReportController.DeleteClubAnnualReport(5);
+            var actual = (result as StatusCodeResult).StatusCode;
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(expected, actual);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+        }
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1258]: unnecessary things

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 10]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var actual = (result as StatusCodeResult).StatusCode;
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(actual, expected);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+        }
+
+        [Test]
+        public async Task EditClubAnnualReport_Invalid_InvalidModelState_Test()
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1283]: Not understandeble name

File: EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs [hunk 11]
```diff
@@ -901,15 +877,474 @@ public async Task CheckCreated_Invalid_UnauthorizedAccessException_Test()
...
+            var actual = (result as StatusCodeResult).StatusCode;
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.AreEqual(expected, actual);
+            Assert.IsInstanceOf<StatusCodeResult>(result);
+        }
+
+
+
```
Comment (MykhailoKlapchuk) [inline EPlast/EPlast.Tests/Controllers/AnnualReportControllerTests.cs:1340]: unnecessary things

**Review by MykhailoKlapchuk — CHANGES_REQUESTED** [review]

**Review by MykhailoKlapchuk — APPROVED** [review]

**Review by LanchevychMaxym — APPROVED** [review]

**Review by AlexKaschuk — APPROVED** [review]

**Pull request merged**